config/config.go

```diff
@@ -22,13 +22,25 @@ import (
 type Install struct {
 	config.Install                `yaml:",inline"`
 	config.Runtime                `yaml:",inline"`
-	Kubeconfig                    string  `yaml:"kube-config,omitempty"`
-	FIFO                          bool    `yaml:"fifo,omitempty"`
-	QPS                           float32 `yaml:"qps,omitempty"`
-	Burst                         int     `yaml:"burst,omitempty"`
-	BinpackAlgo                   string  `yaml:"binpack,omitempty"`
-	InstanceGroupLabel            string  `yaml:"instance-group-label,omitempty"`
-	UseExperimentalHostPriorities bool    `yaml:"use-experimental-host-priorities,omitempty"`
+	Kubeconfig                    string            `yaml:"kube-config,omitempty"`
+	FIFO                          bool              `yaml:"fifo,omitempty"`
+	QPS                           float32           `yaml:"qps,omitempty"`
+	Burst                         int               `yaml:"burst,omitempty"`
+	BinpackAlgo                   string            `yaml:"binpack,omitempty"`
+	InstanceGroupLabel            string            `yaml:"instance-group-label,omitempty"`
+	AsyncClientConfig             AsyncClientConfig `yaml:"async-client-config,omitempty"`
+	UseExperimentalHostPriorities bool              `yaml:"use-experimental-host-priorities,omitempty"`
 
 	ResourceReservationCRDAnnotations map[string]string `yaml:"resource-reservation-crd-annotations,omitempty"`
 }
+
+type AsyncClientConfig struct {
+	maxRetryCount int `yaml:"max-retry-count,omitempty"`
```

Comment:
I think you can use an `*int` here, so you can distinguish not set vs 0 by checking if the value is `nil`, which is the default value for pointers

Reply:
then you can also capitalise the field and access it directly, so you can get rid of the method below

Reply:
@onursatici Updated it to be `*int`, but do you not think it's clearer to keep the method that handles defaulting to 5 in config as a form of documentation and centralising the behaviour?
Kept for now, but happy to change if you think otherwise...